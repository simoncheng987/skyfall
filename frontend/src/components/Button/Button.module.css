.text {
  color: white;
  font-family: 'Poppins';
  font-weight: 900;
  font-size: 24px;
  -webkit-text-stroke: 1px black;
  text-shadow: 0px 2px 2px #1c1c1c;
}

.primary {
  background: #e49359;
  box-shadow: 0 4px 0 0 #783403, 0 5px 0 0 #1c1c1c, 0 6px 6px 0 #1c1c1c;
}

.primary:hover {
  box-shadow: 0 3px 0 0 #783403, 0 4px 0 0 #1c1c1c, 0 4px 5px 0 #1c1c1c;
}

.secondary {
  background: #df5aeb;
  box-shadow: 0 4px 0 0 #600768, 0 5px 0 0 #1c1c1c, 0 6px 6px 0 #1c1c1c;
}

.secondary:hover {
  box-shadow: 0 3px 0 0 #600768, 0 4px 0 0 #1c1c1c, 0 4px 5px 0 #1c1c1c;
}

.grey {
  background: #bdbdbd;
  box-shadow: 0 4px 0 0 #5b5b5b, 0 5px 0 0 #1c1c1c, 0 6px 6px 0 #1c1c1c;
}

.button {
  cursor: pointer;
  min-width: 200px;
  padding: 5px 30px;
  border-width: 1px;
  border-color: #1c1c1c;
  border-radius: 8px;
  transition: box-shadow 50ms, transform 50ms;
}

.button:hover {
  transform: translateY(1px);
}

.button:active {
  transform: translateY(4px);
  box-shadow: none;
}
