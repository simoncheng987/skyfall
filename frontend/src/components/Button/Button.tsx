import React from 'react';
import styles from './Button.module.css';

// Primary (defaut) color is brown.
// Secondary color is pink.
type ButtonStyles = 'primary' | 'secondary';

interface ButtonProps {
  text: string;
  buttonStyle?: ButtonStyles;
  onClick?: React.MouseEventHandler<HTMLButtonElement> | undefined;
  className?: string;
  ariaLabel?: string;
}

export default function Button({ text, buttonStyle, onClick, className, ariaLabel }: ButtonProps) {
  return (
    <button
      aria-label={ariaLabel}
      type="button"
      className={`${className} ${styles.button} ${styles.text} ${styles[buttonStyle as string]}`}
      onClick={onClick}
    >
      {text}
    </button>
  );
}

Button.defaultProps = {
  buttonStyle: 'primary',
  onClick: () => {},
  className: '',
  ariaLabel: '',
};
