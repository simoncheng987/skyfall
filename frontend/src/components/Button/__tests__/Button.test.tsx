import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  const className = 'testClassName';
  const onClick = () => {};
  const buttonText = 'Button Text';

  it('renders the label text prop', () => {
    render(<Button text={buttonText} />);
    expect(screen.getByText(buttonText)).toBeInTheDocument();
  });

  it('simulate click events', () => {
    const testOnClick = jest.fn();

    render(<Button text={buttonText} onClick={testOnClick} />);
    fireEvent.click(screen.getByText(buttonText));
    expect(testOnClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot with no className prop', () => {
    const tree = renderer.create(<Button text={buttonText} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with className prop', () => {
    const tree = renderer.create(<Button className={className} text={buttonText} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with onClick prop', () => {
    const tree = renderer.create(<Button onClick={onClick} text={buttonText} />).toTree();
    expect(tree).toMatchSnapshot();
  });

  it('matches snapshot with buttonStyle prop', () => {
    const tree = renderer.create(<Button buttonStyle="secondary" text={buttonText} />).toTree();
    expect(tree).toMatchSnapshot();
  });
});
