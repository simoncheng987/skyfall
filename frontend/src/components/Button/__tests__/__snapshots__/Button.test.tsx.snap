// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot with buttonStyle prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "ariaLabel": "",
    "buttonStyle": "secondary",
    "className": "",
    "onClick": [Function],
    "text": "Button Text",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "aria-label": "",
      "children": "Button Text",
      "className": " button text secondary",
      "onClick": [Function],
      "type": "button",
    },
    "rendered": Array [
      "Button Text",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;

exports[`Button matches snapshot with className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "ariaLabel": "",
    "buttonStyle": "primary",
    "className": "testClassName",
    "onClick": [Function],
    "text": "Button Text",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "aria-label": "",
      "children": "Button Text",
      "className": "testClassName button text primary",
      "onClick": [Function],
      "type": "button",
    },
    "rendered": Array [
      "Button Text",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;

exports[`Button matches snapshot with no className prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "ariaLabel": "",
    "buttonStyle": "primary",
    "className": "",
    "onClick": [Function],
    "text": "Button Text",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "aria-label": "",
      "children": "Button Text",
      "className": " button text primary",
      "onClick": [Function],
      "type": "button",
    },
    "rendered": Array [
      "Button Text",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;

exports[`Button matches snapshot with onClick prop 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "ariaLabel": "",
    "buttonStyle": "primary",
    "className": "",
    "onClick": [Function],
    "text": "Button Text",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "host",
    "props": Object {
      "aria-label": "",
      "children": "Button Text",
      "className": " button text primary",
      "onClick": [Function],
      "type": "button",
    },
    "rendered": Array [
      "Button Text",
    ],
    "type": "button",
  },
  "type": [Function],
}
`;
